packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModuleConstantsHelper.java

```diff
@@ -191,6 +191,11 @@ private static void validateDirectEventNames(
         }
       }
     }
+    // When providing one event in Kotlin, it will create a SingletonMap by default
+    // which will throw on trying to add new element to it
+    if (events.getClass().getSimpleName().equals("SingletonMap")) {
+      events = new HashMap(events);
+    }
```

Comment:
Consider the logic done below 
```suggestion
    if (!(events instanceof HashMap)) {
      events = new HashMap(events);
    }
```

Reply:
done